internal/pkg/skuba/kubernetes/node_version.go

```diff
@@ -259,7 +259,7 @@ func AllControlPlanesMatchVersionWithVersioningInfo(allNodesVersioningInfo NodeV
 		if !nodeInfo.IsControlPlane() {
 			continue
 		}
-		if !nodeInfo.ToleratesClusterVersion(clusterVersion) {
+		if !nodeInfo.EqualsClusterVersion(clusterVersion) {
```

Comment:
I think this should stay ToleratesClusterVersion should be kept and modified.
Obviously the expectation is not the same as before, and we don't want to tolerate a different patch version, which can be added just after https://github.com/SUSE/skuba/blob/2d340d8b05b0a5a45e567c68333484d9540dd8d4/internal/pkg/skuba/kubernetes/node_version.go#L123
This would allow us to continue with our test coverage.

Also, in that same function https://github.com/SUSE/skuba/blob/2d340d8b05b0a5a45e567c68333484d9540dd8d4/internal/pkg/skuba/kubernetes/node_version.go#L129 might need editing if we don't tolerate the same things for workers now (which is not the scope of this PR).

Reply:
In CaaSP we always shipping fixed and consistent versions of Kubernetes packages. `TolerateClusterVersion` is not checking Patch() version which breaking Patch level upgrade for our case CaaSP v4.2.4

Reply:
Yes that was exactly what I meant :)